cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -69,9 +70,10 @@ func CreateAwsManager(configReader io.Reader) (*AwsManager, error) {
 
 	service := autoscaling.New(session.New())
 	manager := &AwsManager{
-		asgs:     make([]*asgInformation, 0),
-		service:  service,
-		asgCache: make(map[AwsRef]*Asg),
+		asgs:                     make([]*asgInformation, 0),
+		service:                  service,
+		asgCache:                 make(map[AwsRef]*Asg),
+		instancesNotInManagedAsg: make(map[AwsRef]struct{}),
```

Comment:
@andrewsykim Oh, hey, I just realized that we already had the periodic regenerateCache [in our code-base](https://github.com/kubernetes/autoscaler/blob/e9c7cd0/cluster-autoscaler/cloudprovider/aws/aws_manager.go#L80-L86) 😃